### PR TITLE
Leverage Nerdbank.GitVersioning to auto-increment version numbers

### PR DESCRIPTION
**Pull request by tmilnthorp**
Leverage [Nerdbank.GitVersioning](https://github.com/dotnet/Nerdbank.GitVersioning) to auto-increment version numbers

I need your help @angularsen with the NanoFramework items. Not sure if I did it right.

Also I suppose we may need to remove some files (build\bump-version-UnitsNet-minor.bat, etc.)?

### CODE REVIEW

**Comment by angularsen** [comment]
Didn't get to it this weekend, will ping back later.

**Comment by tmilnthorp** [comment]
I think if ADO was not doing a shallow checkout, this should work as-is.

**Comment by tmilnthorp** [comment]
@angularsen are you able to edit the ADO build to do a full depth checkout? (Depth 0)

**Review by angularsen — CHANGES_REQUESTED** [review]
I pushed some changes to match existing branch names and tag names. 
Also some questions/thoughts below.

1. How do we test this without pushing nugets? Should we create a new master branch to simulate how master would be and disable push to nuget.org on that branch?
2. What is the procedure for bumping versions? It seems GitVersioning and their CLI tooling is aimed towards stabilizing releases on new branches, but in UnitsNet it is pretty much about bumping minor versions for every new unit and additive changes. I want to keep it simple, ideally having the CI build bump the minor version for me. Not sure if it can be configured that way, it seems patch version is what is supported for auto-versioning.

File: version.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "publicReleaseRefSpec": [
+    "^refs/heads/master$",
+    "^refs/heads/v\\d+(?:\\.\\d+)?$"
+  ],
+  "cloudBuild": {
+    "buildNumber": {
+      "enabled": true
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (tmilnthorp) [reply version.json]: You can create nested version.json files if we want to split it still, and set a nbgv path filter

**Comment by tmilnthorp** [comment]
> How do we test this without pushing nugets? Should we create a new master branch to simulate how master would be and disable push to nuget.org on that branch?

1. Make sure you have nbgv installed locally: ```dotnet tool install -g nbgv```
2. Checkout ```master``` locally and merge in this branch
3. Run ```nbgv get-version```
4. ```git reset origin/master``` to reset local master

> What is the procedure for bumping versions? It seems GitVersioning and their CLI tooling is aimed towards stabilizing releases on new branches, but in UnitsNet it is pretty much about bumping minor versions for every new unit and additive changes. I want to keep it simple, ideally having the CI build bump the minor version for me. Not sure if it can be configured that way, it seems patch version is what is supported for auto-versioning.

Hmmm indeed. Let me see if I can make it get the minor to increment

**Pull request closed without merging**